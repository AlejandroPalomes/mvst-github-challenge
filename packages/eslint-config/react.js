const { resolve } = require("node:path");

const project = resolve(process.cwd(), "tsconfig.json");

/*
 * This is a custom ESLint configuration for use a library
 * that utilizes React.
 *
 * This config extends the Vercel Engineering Style Guide.
 * For more information, see https://github.com/vercel/style-guide
 *
 */

module.exports = {
  extends: [
    "@vercel/style-guide/eslint/browser",
    "@vercel/style-guide/eslint/typescript",
    "@vercel/style-guide/eslint/react",
  ].map(require.resolve),
  parserOptions: {
    project,
  },
  plugins: ["only-warn"],
  globals: {
    JSX: true,
  },
  settings: {
    "import/resolver": {
      typescript: {
        project,
      },
    },
  },
  ignorePatterns: ["node_modules/", "dist/", ".eslintrc.js", "**/*.css"],
  // add rules configurations here
  rules: {
    "import/no-default-export": "off",
    "@typescript-eslint/explicit-function-return-type": "off",
    "jsx-a11y/click-events-have-key-events": "off",
    "no-console": "off",
    "@typescript-eslint/no-unsafe-member-access": "off",
    "@typescript-eslint/no-explicit-any": "off",
    "react/jsx-no-leaked-render": "off",
    "jsx-a11y/no-static-element-interactions": "off",
    "react/function-component-definition": [2, { "namedComponents": "arrow-function" }]
  },
};
