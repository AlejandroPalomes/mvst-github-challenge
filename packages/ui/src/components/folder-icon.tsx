import { type FC } from 'react';
import { type IconProps } from '../utils/icon-types';
import { getStrokeWidth } from '../utils/icon-helper';

export const FolderIcon: FC<IconProps> = ({ color = 'white', size = 24}) => {
  const strokeWidth = getStrokeWidth(size);

  return <svg
    fill="none"
    height={size}
    viewBox="0 0 24 24"
    width={size}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13.0072 6.19808L11.7749 4.91872C11.2094 4.33167 10.4294 4 9.61425 4H5.5C4.39543 4 3.5 4.89543 3.5 6V8.3072M13.0072 6.19808H18.6634C19.768 6.19808 20.6634 7.09351 20.6634 8.19808V19.1858C20.6634 19.738 20.2157 20.1858 19.6634 20.1858H4.5C3.94771 20.1858 3.5 19.738 3.5 19.1858V8.3072M13.0072 6.19808L10.4359 8.10972C10.2634 8.23795 10.0542 8.3072 9.83929 8.3072H3.5"
      stroke={color}
      strokeLinecap="round"
      strokeWidth={strokeWidth}
    />
  </svg>
  ;
};
