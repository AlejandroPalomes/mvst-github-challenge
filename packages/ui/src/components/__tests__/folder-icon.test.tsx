import { render } from "@testing-library/react";
import { FolderIcon } from "../folder-icon";

describe('FolderIcon test', () => {
  it('should render without crashing', () => {
    const { baseElement } = render(<FolderIcon/>);
    expect(baseElement).toMatchSnapshot();
  });
})