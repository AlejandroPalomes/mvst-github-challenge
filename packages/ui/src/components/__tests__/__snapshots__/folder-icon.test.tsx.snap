// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FolderIcon test > should render without crashing 1`] = `
<body>
  <div>
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.0072 6.19808L11.7749 4.91872C11.2094 4.33167 10.4294 4 9.61425 4H5.5C4.39543 4 3.5 4.89543 3.5 6V8.3072M13.0072 6.19808H18.6634C19.768 6.19808 20.6634 7.09351 20.6634 8.19808V19.1858C20.6634 19.738 20.2157 20.1858 19.6634 20.1858H4.5C3.94771 20.1858 3.5 19.738 3.5 19.1858V8.3072M13.0072 6.19808L10.4359 8.10972C10.2634 8.23795 10.0542 8.3072 9.83929 8.3072H3.5"
        stroke="white"
        stroke-linecap="round"
        stroke-width="1.5"
      />
    </svg>
  </div>
</body>
`;
